### ISSUE TITLE
Last Sign-In not set at user creation.

### ISSUE THREAD

**Issue by JulienPalard** [go-gitea/gitea#21698]
### Description

In `admin/users` I see users with `Last Sign-In` set to `Never Signed-In` while they obviously kind of "signed in" (one of them at least created an organization).

It feel wrong that a user that "never signed-in" have created an organization.

So to me it feel like the "last sign-in" should be set to the current date when the user is "automatically signed in" at account creation.

### Gitea Version

1.17.3

### Can you reproduce the bug on the Gitea demo site?

No (need to be admin, I don't think I'll get admin there).

### Operating System

Debian

### How are you running Gitea?

Self-hosted instance, installed from https://dl.gitea.io, started using systemd.

### Database

PostgreSQL

**Comment by wolfogre** [comment]
I can't reproduce the bug. Actually, the "last sign-in" has already been set to the current date after account creation. I checked the previous versions, and it always works like that since the "auto sign in" has been added at #4615.

I think it may be caused by something else.

**Comment by JulienPalard** [comment]
Strange. I don't know how I can diagnose this though. Out of 11 users 5 of them are marked "Never Signed-In", one of them has created an org, but none of them has created a repo.

**Comment by techknowlogick** [comment]
Thanks for this ticket.

I'm able to see this behaviour on try.gitea.io (as an admin).

<img width="419" alt="a user with one repo, a creation date, but no sign in date" src="https://user-images.githubusercontent.com/164197/200715305-dcf34596-f1fa-4c72-b1b4-6c809ff93655.png">

@JulienPalard What is the setup for your instance, is it public, do you require email confirmation for signups, or any other info you may think might be helpful.

**Comment by JulienPalard** [comment]
> @JulienPalard What is the setup for your instance, is it public, do you require email confirmation for signups, or any other info you may think might be helpful.

Our setup is fully public here: https://github.com/AFPy/infra/blob/main/gitea.yml and here : https://github.com/AFPy/infra/tree/main/roles/gitea (browse the `tasks` and `templates` folders if you're not acustomed to Ansible).

TL;DR we have `REGISTER_EMAIL_CONFIRM            = true`.

**Comment by JulienPalard** [comment]
Oh I'm late to the party :)

Thanks taking the time to understand the bug and thanks for fixing it!!!
